### PR TITLE
Add explanatory text for error E0599

### PR DESCRIPTION
**Pull request by neunenak**
Add a text description of this error instead of just example error code

r? GuillaumeGomez

### CODE REVIEW

**Comment by neunenak** [comment]
I think I was supposed to mention @GuillaumeGomez specifically and messed up the notification syntax, sorry about that.

File: src/librustc_typeck/diagnostics.rs [hunk 1]
```diff
@@ -3986,6 +3986,9 @@ details.
 "##,
 
 E0599: r##"
+
```
Comment (GuillaumeGomez) [inline src/librustc_typeck/diagnostics.rs]: Please remove this line.

File: src/librustc_typeck/diagnostics.rs [hunk 2]
```diff
@@ -3986,6 +3986,9 @@ details.
...
+This error occurs when a method is used on a type which doesn't implement it:
```
Comment (GuillaumeGomez) [inline src/librustc_typeck/diagnostics.rs]: Please update this to:

```
This error occurs when a method is used on a type which doesn't implement it.

Erroneous code example:
```

**Comment by GuillaumeGomez** [comment]
Once updated, it's good to go. Thanks!

**Comment by neunenak** [comment]
Updated this commit per comments.

**Pull request merged**